templates/kube-oidc-proxy.yaml

```diff
@@ -0,0 +1,55 @@
+apiVersion: kubeaddons.mesosphere.io/v1alpha1
+kind: Addon
+metadata:
+  name: kube-oidc-proxy
+  namespace: kubeaddons
+  labels:
+    kubeaddons.mesosphere.io/name: kube-oidc-proxy
+  annotations:
+    appversion.kubeaddons.mesosphere.io/dex-k8s-authenticator: "v1.1.0"
+    values.chart.helm.kubeaddons.mesosphere.io/dex-k8s-authenticator: "https://raw.githubusercontent.com/mesosphere/charts/f33a09a3db1005d0c2facbd4c115a3e5d2fba9e5/staging/kube-oidc-proxy/values.yaml"
+spec:
+  kubernetes:
+    minSupportedVersion: v1.15.0
+  cloudProvider:
+    - name: aws
+      enabled: true
+    - name: docker
+      enabled: true
+    - name: none
+      enabled: true
+  chartReference:
+    chart: kube-oidc-proxy
+    repo: https://mesosphere.github.io/charts/staging
+    version: 0.1.1
+    values: |
+      ---
+      image:
+        repository: quay.io/jetstack/kube-oidc-proxy
+        tag: v0.1.1
+        pullPolicy: IfNotPresent
+
+      ingress:
+        enabled: true
+        annotations:
+          kubernetes.io/ingress.class: traefik
+          ingress.kubernetes.io/protocol: https
+        path: /oidc
+        hosts:
+          - ""
+
+      # Created by init-container
+      tls:
+        secretName: kube-oidc-proxy-kubeaddons-server-tls
+
+      oidc:
+        clientId: kube-apiserver
+        # updated by initcontainer when working with a remote dex
+        issuerUrl: https://dex-kubeaddons.kubeaddons.svc.cluster.local:8080/dex
+        usernameClaim: email
+        # placeholder : updated by initcontainer
+        caPEM: "placeholder"
+      initContainers:
+      - name: initialize-kube-oidc-proxy
+        image: jaredrodriguez/kubeaddons-addon-initializer:latest
```

Comment:
let's use the our shared docker hub repo with a static tag, `latest` is a bit dangerous for actual deployments

Reply:
+1

Reply:
@s12chung can you tag a new version of kubeaddons-addon-initializer with the latest changes ?

Reply:
building...

Reply:
Thanks, this was here only for testing.